playwright-e2e/dsm/pages/miscellaneous-pages/user-and-permissions-page.ts

```diff
@@ -0,0 +1,307 @@
+    const button = this.getAddUserButton();
+    await expect(button).toBeVisible();
+  }
+
+  public async assertStudyAdminInfo(email: string, name: string, phone?: string): Promise<void> {
+    //Get the relevant study admin's section
+    const studyAdmin = this.getStudyAdmin(email);
+
+    //Verify the email is as expected
+    const studyAdminEmail = await studyAdmin.innerText();
+    expect(studyAdminEmail).toBe(email);
+
+    //Verify the name is as expected
+    const studyAdminName = await this.getStudyAdminName(name);
+    expect(studyAdminName).toBe(name);
+
+    if (phone) {
+      //If phone information is provided, verify that it is shown in the page as expected
+      const studyAdminPhoneNumber = await this.getStudyAdminPhoneNumber(phone);
+      expect(studyAdminPhoneNumber).toBe(phone);
+    }
+  }
+
+  public async assertAllPossibleStudyPermissionsAreVisible(study: StudyEnum, email: string): Promise<void> {
+    //Verify that all possible permissions for the study are available for the study admin to select
+    const studyAdmin = this.getStudyAdmin(email);
+    const allPossiblePermissions = await this.getPermissionsSection(studyAdmin);
+    const amountOfPermissions = allPossiblePermissions.length;
+    console.log(`Amount of permissions: ${amountOfPermissions}`);
+
+    for (let index = 0; index < amountOfPermissions; index++) {
+      const permissionCheckbox = allPossiblePermissions[index];
+      const permission = permissionCheckbox.locator(`//span[@class='ng-star-inserted']`);
+      const currentPermission = (await permission.innerText()).trim();
+      //console.log(`Current permission: ${currentPermission}`);
+      const currentPermissionCastedAsEnum = currentPermission as UserPermission;
+      const verifiedPermission = Object.values(this.currentStudyPermissions).indexOf(currentPermissionCastedAsEnum) >= 0;
+      if (!verifiedPermission) {
+        throw new Error(`The ${currentPermission} permission is not a valid permission for ${study}`);
+      }
+    }
+  }
+
+  public async assertSelectedPermissions(studyAdmin: Locator, permissions: UserPermission[]): Promise<void> {
+    const amountOfPermissionstoCheckFor = permissions.length;
+
+    for (let index = 0; index < amountOfPermissionstoCheckFor; index++) {
+      const checkbox = this.getPermissionCheckbox(studyAdmin, permissions[index]);
+      await expect(checkbox).toBeChecked();
+    }
+  }
+
+  private getPermissionCheckbox(studyAdmin: Locator, permission: UserPermission, forSelection?: boolean): Locator {
+    const permissionName = permission as string;
+    console.log(`Permission Name: ${permissionName}`);
+    if (forSelection) {
+      //below xpath works best for selection/clicking
+      return studyAdmin.locator(`//ancestor::mat-expansion-panel//app-permission-checkbox[contains(.,'${permissionName}')]//div`)
+    }
+    //below xpath works best for determining if the checkbox is currently selected
```

Comment:
We probably don't need the "works best" comments here since there are lots of other uses of xpath in this page class.  Might be worth a comment on the class itself noting why xpath is preferred here.  If the rest of our tests rely on xpath, we probably don't need a comment about it at all.